### PR TITLE
Add a draft proposal of requirements.

### PR DESCRIPTION
**Pull request by dajiaji**

### CODE REVIEW

**Comment by tomoyukilabs** [comment]
Great! Thanks, @dajiaji! I'll add several comments inline. PTAL.

First, I would like to merge this PR after you check my comments. Then, I would like to suggest several modifications by submitting another PR.

File: UseCases.md [hunk 1]
```diff
@@ -63,3 +63,52 @@ if she usually posts photos to the online service from her smartphone directly,
 
 ![uc06](figs/uc06.jpg)
 
+# Use Case Analysis
+
+- #T.B.D.
+    - Network environment: a local network and/or a global network
+    - Certificate issuer: public CA / corporate or organizational CA / private CA
+    - Privacy scope: public / per service or device manufacturer / private
```
Comment (tomoyukilabs) [inline UseCases.md]: We can omit this section for now, IMO.

Comment (dajiaji) [inline UseCases.md]: Thank you for your review. Okay, you can omit the section.

File: UseCases.md [hunk 2]
```diff
@@ -63,3 +63,52 @@ if she usually posts photos to the online service from her smartphone directly,
...
+# Requirements for HTTPS/WSS in Local Network
+
+This section collects requirements derived from use cases listed above.
+
+## <a name="req-01"></a>REQ-01: Device Discovery
+
+- The UA (the web browser mentioned in the use cases above) shall be able to securely discover the presence of HTTPS/WSS server capable devices (hereinafter just called 'device') that are connected to the local network. 
+- A secure context loaded from the internet to the UA (hereinafter just called 'secure context') should also be able to discover target device capabilities that are actively (e.g., turned on) connected to the local network (e.g., device type, identity of a set of Web APIs, and so on).
+- A secure context shall be able to get access to the locally discovered device based on the user consent.
+- If there are multiple devices in local network, the UA shall be able to provide the user with a way to select one device at a time which she intends to use on the secure context.
```
Comment (dajiaji) [inline UseCases.md]: > Should we limit network scope of devices to the same local network?

I'd like to focus on the case to simplify our discussion if  there isn't anyone who wants to consider more complicated local network environment.

> Should the UA provide a way to limit devices to ones which has capabilities requested by the users?

I'm not sure but yes, I think so. We need more discussion but  you can add the requirement. 

> Should the UA avoid exposing the device list to web apps?

Yes. Please add the requirement.

File: UseCases.md [hunk 3]
```diff
@@ -63,3 +63,52 @@ if she usually posts photos to the online service from her smartphone directly,
...
+	
+## <a name="req-02"></a>REQ-02: Mutual authentication between device and secure context
+
+- The secure context must have a way to verify whether the device to which it tries getting access is reliable or not.
+- The device should have a way to verify whether the origin of the secure context which tries getting access to the device is reliable or not.
+- etc. 
+
+## <a name="req-03"></a>REQ-03: Issuing TLS server certificate for device
+
+(Are there any solution to realize the use cases above without issuing a TLS server certificate to the device ?)
```
Comment (dajiaji) [inline UseCases.md]: Okay. Thanks.

File: UseCases.md [hunk 4]
```diff
@@ -63,3 +63,52 @@ if she usually posts photos to the online service from her smartphone directly,
...
+
+- The device must have a way to get a server certificate which the UA can trust after connecting to the local network because an IP address and a domain name of a device in local network is changeable.
```
Comment (dajiaji) [inline UseCases.md]: Thanks. I've modified the draft. Can you please check and merge it ?

File: UseCases.md (hunk 2 above)
Comment (tomoyukilabs) [inline UseCases.md]: Okay, I'll merge this as it is for now. These items could be discussed later in https://github.com/httpslocal/usecases/issues/4.

**Comment by dajiaji** [comment]
Thanks for merging it.

**Pull request merged**